### PR TITLE
chore(dal, sdf): Refactoring the Asset Authoring experience behind a client

### PR DESCRIPTION
**Pull request by stack72**
This keeps it inline with Func Authoring and will allow us to test the pieces in isolation rather than needing to go through SDF for testing

### CODE REVIEW

**Comment by stack72** [comment]
bors merge

File: lib/dal/src/schema/variant/authoring.rs [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+type VariantAuthoringResult<T> = Result<T, VariantAuthoringError>;
+
+const DEFAULT_ASSET_CODE: &str = r#"function main() {
+  return new AssetBuilder().build()
+}"#;
+
+#[derive(Debug)]
+pub struct VariantAuthoringClient;
+
+impl VariantAuthoringClient {
```
Comment (nickgerace) [inline lib/dal/src/schema/variant/authoring.rs:64]: Oh so beautiful :sob:

**Review by nickgerace — APPROVED** [review]

**Comment by nickgerace** [comment]
bors merge

**Pull request merged**